### PR TITLE
Add opencv config property

### PR DESCRIPTION
**Pull request by bf-cortex**

### CODE REVIEW

**Comment by giacomocerquone** [comment]
hi @bf-cortex do you have a link where to see all the possible config properties one might want to put in there?

**Comment by bf-cortex** [comment]
hi @giacomocerquone unfortunately no but you can look in this file for more Module[] variables:
https://docs.opencv.org/4.5.3/opencv.js

I think that these might be different from version to version and we just need only a way to pass dynamic variables to opencv.
I also observed today that when I build opencv from source, window.cv is a promise when onRuntimeInitialized is triggered.

In opencv's utils.js, the onload event checks if it is a promise:
https://docs.opencv.org/4.5.3/utils.js

**Comment by giacomocerquone** [comment]
> hi @giacomocerquone unfortunately no but you can look in this file for more Module[] variables:
> https://docs.opencv.org/4.5.3/opencv.js
> 
> I think that these might be different from version to version and we just need only a way to pass dynamic variables to opencv.
> I also observed today that when I build opencv from source, window.cv is a promise when onRuntimeInitialized is triggered.
> 
> In opencv's utils.js, the onload event checks if it is a promise:
> https://docs.opencv.org/4.5.3/utils.js

Even if it's a promise, I'd say that it's up to the user to await it or not and check if it's a promise.§

So, a dev would write something like this:

```
const App = () => {
  const onLoaded = async (cv) => {
    console.log('opencv loaded', await cv)
  }

  return (
    <OpenCvProvider onLoad={onLoaded} openCvPath='/opencv/opencv.js'>
      <MyComponent />
    </OpenCvProvider>
  )
}
```

**Comment by giacomocerquone** [comment]
So I'd like to not handle it inside the library. Instead I'm good with the module config injection :)

**Comment by bf-cortex** [comment]
Yes, that's how I tried it first but then it always returns a promise from the useOpenCv hook. I think it's prettier to check it directly in the library.

**Comment by giacomocerquone** [comment]
@bf-cortex if you still want to work on this I'd give it a go!
For start, I'd solve the conflicts (should be very easy, I've just merged another guy's pr)

File: src/lib/OpenCvProvider.js [hunk 1]
```diff
@@ -46,7 +49,7 @@ export const OpenCvProvider = ({ openCvPath, children, onLoad }) => {
       return js
     }
 
-    document.body.appendChild(generateOpenCvScriptTag())
+    generateOpenCvScriptTag().then(el => document.body.appendChild(el))
```
Comment (giacomocerquone) [inline src/lib/OpenCvProvider.js:52]: there is definitely no need for a promise here

File: src/lib/OpenCvProvider.js [hunk 2]
```diff
@@ -32,12 +35,12 @@ export const OpenCvProvider = ({ openCvPath, children, onLoad }) => {
...
     moduleConfig.onRuntimeInitialized = handleOnLoad
-    window.Module = moduleConfig
+    window.Module = { ...moduleConfig, ...openCvConfig }
 
-    const generateOpenCvScriptTag = () => {
+    const generateOpenCvScriptTag = async () => {
       const js = document.createElement('script')
       js.id = scriptId
-      js.src = openCvPath || 'https://docs.opencv.org/3.4.13/opencv.js'
+      js.src = (typeof openCvPath === 'function') ? await openCvPath() : openCvPath || 'https://docs.opencv.org/3.4.13/opencv.js'
```
Comment (giacomocerquone) [inline src/lib/OpenCvProvider.js:43]: why would you pass an async function as the openCvPath param which should be a string? Doesn't make sense to asynchronously wait for a string. What's/was your use case?

**Pull request closed without merging**